controllers/nc-news-controlers.js

```diff
@@ -74,14 +75,36 @@ function postCommentByArticleId(request, response, next) {
           status: 404,
           msg: `The article_id provided does not exist`,
         });
-      } else {
-        return addCommentByArticleById(article_id, commentData);
       }
+      return addCommentByArticleById(article_id, commentData);
     })
     .then((comment) => {
       response.status(201).send({ comment });
     })
     .catch((err) => {
+      //console.log(err);
+      next(err);
+    });
+}
+
+function patchArticleByArticleId(request, response, next) {
+  const { article_id } = request.params;
+  const artData = request.body;
+  fetchArticleById(article_id)
+    .then((article) => {
+      if (article.length === 0) {
+        return Promise.reject({
+          status: 404,
+          msg: `The article_id provided does not exist`,
+        });
```

Comment:
You want to have the promise.reject in your models 👀 this ensures that an invalid article ID is caught early, making your code cleaner and easier to manage 🧹